web/src/main/webapp/js/translate.js

```diff
@@ -6,11 +6,11 @@ var mathTools = require('./tools/math.js');
 
 function tr2(key, args) {
     if (key === null) {
-        log("ERROR: key was null?");
+        console.log("ERROR: key was null?");
         return "";
     }
     if (defaultTranslationMap === null) {
-        log("ERROR: defaultTranslationMap was not initialized?");
```

Comment:
Correct, log() is undefined. There are more instances, but are all commented out, so I did not fix those.